### ISSUE TITLE
Parallel restore for same repo and same local cache throws a lot of non-fatal errors

### ISSUE THREAD

**Issue by whereisaaron** [restic/restic#2219]
Output of `restic version`
--------------------------
restic 0.9.4 compiled with go1.11.4 on linux/amd64

How did you run restic exactly?
-------------------------------
```
restic --repo b2:foo restore latest --include '*/foo/*' --target . &
restic --repo b2:foo restore latest --include '*/bar/*' --target . &
```

What backend/server/service did you use to store the repository?
----------------------------------------------------------------
B2

Expected behavior
-----------------
Two restore run in parallel without error.

Actual behavior
---------------

Running restores is parallel against the same repo and using the same local cache I get a lot of these errors, though hopefully not fatal, rather two restics both try to update the local cache.

Load(<snapshot/468a863ff7>, 0, 0) returned error, retrying after 2.563959855s: Create: open /var/foo/restic/541ced1298f269c45308c46e8849e5624c0c061b5649597445ca538405e47260/snapshots/46/468a863ff73c0e1ae6a1ec8dc2d2c8dfe11c8e25b85a015f00f8e45433a9e598: file exists

Steps to reproduce the behavior
-------------------------------

```
restic --repo b2:foo restore latest --include '*/foo/*' --target . &
restic --repo b2:foo restore latest --include '*/bar/*' --target . &
```

**Comment by MichaelEischer** [comment]
Closing this as duplicate of #2345, where this has been discussed in detail. Essentially it's a race condition between two restic clients writing to the same cache directory.
